hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -2893,6 +2907,7 @@ protected void setDefaults() {
           HoodieWriteCommitCallbackConfig.newBuilder().fromProperties(writeConfig.getProps()).build());
       writeConfig.setDefaultOnCondition(!isPayloadConfigSet,
           HoodiePayloadConfig.newBuilder().fromProperties(writeConfig.getProps()).build());
+      setMetadataConfigsSetByUser();
```

Comment:
we should call this before calling 
```
writeConfig.setDefaultOnCondition(!isMetadataConfigSet,
          HoodieMetadataConfig.newBuilder().withEngineType(engineType).fromProperties(writeConfig.getProps()).build());
```
if not, default value from code will be set in properties map and we can't deduce if something was explicitly set by the user or was it set as default by code.